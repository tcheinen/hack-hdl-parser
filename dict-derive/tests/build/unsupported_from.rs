use dict_derive::FromPyObject;

struct Test {
    test: String,
}

#[derive(FromPyObject)]
struct Derived {
    test: Test,
}

fn main() {}
