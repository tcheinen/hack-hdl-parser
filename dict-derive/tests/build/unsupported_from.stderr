error[E0277]: the trait bound `Test: PyClass` is not satisfied
 --> $DIR/unsupported_from.rs:9:11
  |
7 | #[derive(FromPyObject)]
  |          ------------ required by this bound in `extract_required`
8 | struct Derived {
9 |     test: Test,
  |           ^^^^ the trait `PyClass` is not implemented for `Test`
  |
  = note: required because of the requirements on the impl of `FromPyObject<'_>` for `Test`

error[E0277]: the trait bound `Test: Clone` is not satisfied
 --> $DIR/unsupported_from.rs:9:11
  |
7 | #[derive(FromPyObject)]
  |          ------------ required by this bound in `extract_required`
8 | struct Derived {
9 |     test: Test,
  |           ^^^^ the trait `Clone` is not implemented for `Test`
  |
  = note: required because of the requirements on the impl of `FromPyObject<'_>` for `Test`
